Bug: Enum constants in when/is fail type check when variable is explicitly typed as int

## Description

When an enum value is stored in a variable with an explicit `int` type annotation, using enum constants in `when/is` cases produces a type mismatch error (E3001). However, the same comparison works fine with the `==` operator.

## Steps to Reproduce

```ez
import @std
using std

const Direction enum {
    NORTH
    EAST
    SOUTH
    WEST
}

do main() {
    temp dir int = Direction.SOUTH
    when dir {
        is Direction.NORTH { println("north") }
        is Direction.SOUTH { println("south") }
        default { println("other") }
    }
}
```

## Expected Behavior

The `when/is` should match `Direction.SOUTH` against the `int` variable since integer enums evaluate to `int` values.

## Actual Behavior

```
error[E3001]: case value type Direction does not match when value type int
```

## Workarounds

1. Use type inference: `temp dir = Direction.SOUTH` (inferred type is `Direction`, which works)
2. Use raw int values: `is 2 { ... }`
3. Use `if`/`==` instead: `if dir == Direction.SOUTH { ... }` (works fine)

## Notes

- The `==` operator correctly handles `int == Direction.X` comparisons, so the inconsistency is specifically in the `when/is` typechecker logic.
- This is a common pattern users would write, especially when receiving enum values from functions that return `int`.

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -4651,7 +4651,12 @@ func (tc *TypeChecker) checkWhenStatement(whenStmt *ast.WhenStatement, expectedR
 				// Allow enum type matching
 				if !strings.HasPrefix(caseType, valueType) && !strings.HasSuffix(caseType, "."+valueType) {
 					// Allow int case values when matching against enum types (enums have int underlying values)
-					if !(isEnumType && caseType == "int") {
+					// Also allow enum case values when matching against int (e.g., temp dir int = Direction.SOUTH)
+					caseIsEnumWithIntBase := false
+					if caseEnumInfo, isCaseEnum := tc.GetType(caseType); isCaseEnum && caseEnumInfo.Kind == EnumType {
+						caseIsEnumWithIntBase = caseEnumInfo.EnumBaseType == valueType
+					}
+					if !(isEnumType && caseType == "int") && !caseIsEnumWithIntBase {
 						line, col := tc.getExpressionPosition(caseValue)
 						tc.addError(
 							errors.E3001,

